src/main/kotlin/com/hypto/iam/server/apis/OrganizationApi.kt

```diff
@@ -28,17 +33,33 @@ import org.koin.ktor.ext.inject
  * API to create organization in IAM.
  */
 fun Route.createOrganizationApi() {
-    val service: OrganizationsService by inject()
+    val organizationService: OrganizationsService by inject()
+    val passcodeRepo: PasscodeRepo by inject()
     val gson: Gson by inject()
 
     post("/organizations") {
-        val request = call.receive<CreateOrganizationRequest>().validate()
-        val passcodeStr = call.principal<ApiPrincipal>()?.tokenCredential?.value
-        val (organization, tokenResponse) = service.createOrganization(
-            request.name,
-            description = request.description ?: "",
-            rootUser = request.rootUser,
-            passcodeStr = passcodeStr
+        val passcodeStr = call.principal<ApiPrincipal>()?.tokenCredential?.value!!
+
+        val apiRequest = call.receiveOrNull<CreateOrganizationRequest>()
+        val passcodeMetadata = passcodeRepo.getValidPasscode(passcodeStr, VerifyEmailRequest.Purpose.signup)?.metadata
+        if (passcodeMetadata != null && apiRequest != null) {
+            throw BadRequestException("Organization and root user details are already provided")
```

Comment:
nit: Update the error message

"Admin user details are provided both during passcode creation time and sign-up request."